app/src/main/java/com/example/indoor_volleyball/Activities/CreateEventActivity.java

```
@@ -126,12 +143,42 @@ public void onClick(View v) {
                 } catch (ParseException e) {
                     e.printStackTrace();
                 }
-                thisGym.setNextEvent(nextEvent);
-                thisGym.saveInBackground();
             }
         });
     }
 
+    public void setAllowPlusOnesSpectators() {
+        if (binding.cbAllowSpectators.isChecked()) {
+            allowSpectators = true;
+        }
+        if (binding.cbAllowPlusOnes.isChecked()) {
+            allowPlusOnes = true;
+        }
+    }
+
+    //TODO buggy sends to both accounts I think due to login history.
+    public void sendNotification(Gym gym) throws ParseException {
+        ParseQuery<ParseUser> userQuery = ParseUser.getQuery();
+        userQuery.whereEqualTo("gymsFollowing", gym);
+        // Find devices associated with these users
+        ParseQuery<ParseInstallation> pushQuery = ParseInstallation.getQuery();
+        pushQuery.whereMatchesQuery("user", userQuery);
+        JSONObject data = new JSONObject();
+        // Put data in the JSON object
+        try {
+            data.put("alert", "New event created at " + gym.getName());
+            data.put("title", "New Event!");
+        } catch (JSONException e) {
+            // should not happen
+            throw new IllegalArgumentException("unexpected parsing error", e);
```

I don't think you want to throw a different exception.  Probably have the method throw a JSONException (I don't think it needs to throw ParseException)

---

Or throw both if it does need parse exception.  You're right though, you should never hit that, you're jsut adding things to a new object.